src/features/filter.ts

```diff
@@ -1,12 +1,38 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
+import { RootState } from '../app/store';
 
 const initialState = {
   query: '',
-  status: 'all',
+  status: Status.All,
 };
 
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setStatusFilter: (state, action: PayloadAction<Status>) => {
+      return {
+        ...state,
+        status: action.payload,
+      };
+    },
+    setSearchQuery: (state, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        query: action.payload,
+      };
+    },
+    resetFilters: () => {
+      return initialState;
```

Comment:
Ensure that the `initialState` is returned correctly in the `resetFilters` reducer. While returning `initialState` directly is generally safe, be cautious of any future changes to `initialState` that might inadvertently mutate the state.